### PR TITLE
Cleaned up tos / privacy policy link usage

### LINKED ISSUE #783: Fix terms of service links and clean up old content
Most of our terms of service links still go to `/terms_of_service`, which is a URL that serves a static Django template with our old TOS. Our new place for up-to-date TOS is the CMS, with the path `/terms-of-service`. We need to update the deprecated links, clean up the old content/url routes, and make sure that any templates with hardcoded links to resource pages are updated to use context variables for those URLs.

### Acceptance criteria
- `main.views.get_base_context` updated to return the up-to-date terms of service link (`/terms-of-service`) in the `resource_page_urls` dict
-  All existing `/terms_of_service` links (Django route name: `bootcamp-tos`) updated to use the terms of service link provided by `main.views.get_base_context`
- `main/templates/bootcamp/tos.html` and `main/templates/bootcamp/tac.html` deleted, along with any route definitions, tests, etc.
- Any other hardcoded resource page links in Django templates (ex: `/privacy-policy`) updated to use the equivalent from `main.views.get_base_context`
  - **NOTE:** If there are templates that need the resource page URLs, but don't need all of the other context variables in `main.views.get_base_context`, then the resource page URLs should be returned from a separate function, and those templates can receive those resource page URLs by calling that new function.

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### What are the relevant tickets?
Fixes #783 

#### What's this PR do?

This PR covers a fair amount of cleanup around TOS and related links:
- Add a custom setting page (`SitePageSettings`) to support configuring which pages are TOS, Privacy Policy, etc
- Rolled `main.views.get_base_context` into `main.context_processors` so that these get automatically added to all views (we were manually doing this in all views before).
- Cleaned up the email verification template, which had a duplicate (but more correct) footer in it

#### How should this be manually tested?

- Login to the CMS, there should be a "Resource Pages" page under "Settings"
  - For each of the options listed, create a page of the `ResourcePage` type and associate it on this page
  - Save the new settings
- Poke about the app, including logging into `/applications` and verify the context processor changes didn't break any existing views.
- Verify the links in the footer go where you'd expect them to
- Go to `/__emaildebugger__/` and verify for each of the emails the footer links render correctly
  - Also verify the fixes for the verification template noted above
- Create a Letter template page and verify the preview link still works

#### Any background context you want to provide?

A simpler approach to this would've been to just update the current hardcoded links, I opted for the custom setting approach as that seemed a) more flexible, leaving the url/slug up to CMS users and b) more idiomatic to how Wagtail likes to set things up. It meant a bit of extra work, but it means we'll never have to update these urls in the future if a different path is desired. It also sets a best-practice for how we represent these in the app.

### CODE REVIEW

File: cms/utils.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    """
+    Get resource page urls for a given site
+
+    Args:
+        site(wagtail.models.Site): the site to get resource pages for
+
+    Returns:
+        dict: the set of resource pages
+    """
+    site_page_settings = ResourcePagesSettings.for_site(site)
```
[the remark this answers is not in the document]
Comment (umarmughal824) [reply cms/utils.py:15]: I guess it's being used only for email context but for the rest of all the views the context processor is being used, so it should be the issue.

[the remark this answers is not in the document]
Comment (umarmughal824) [reply cms/utils.py:15]: @gsidebo I think that function is only being used for email context but for the rest of all the views context processor is being used. So it. should be the issue here.

File: cms/utils.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    pages = {
+        "how_to_apply": site_page_settings.apply_page,
+        "about_us": site_page_settings.about_us_page,
+        "bootcamps_programs": site_page_settings.bootcamps_programs_page,
+        "privacy_policy": site_page_settings.privacy_policy_page,
+        "terms_of_service": site_page_settings.terms_of_service_page,
+    }
+    print(pages)
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply cms/utils.py:23]: My kryptonite!

**Comment by rhysyngsun** [comment]
@gsidebo @umarmughal824 I've added the caching using the same library we're using in Open (so we know it's solid). It'll cache for an hour or invalidate if the settings or any resource page is changed.

**Pull request merged**